File: PHP/Objects/AndroidApp.php
<?php
use Facebook\WebDriver;
use Facebook\WebDriver\WebDriverBy;

class ApiDemos {
  private $driver_, $appCategoryXpath;

  function __construct($driver) {
    $this->driver_ = $driver;
    $this->appCategoryXpath = "//android.widget.TextView[@content-desc='App']";
  }

  public function clickOnAppCategory() {
    try {
      $this->driver_->findElement(WebDriverBy::xpath($this->appCategoryXpath))->click();
      return true;
    }
    catch(Exception $ex){
      echo "ERROR: ", $ex->getMessage(), "\n";
      return false;
    }
  }
}
?>

File: Ruby/Objects/AndroidApp.rb
require 'selenium-webdriver'
require_relative '../Utils/Driver'
require_relative '../Utils/Helpers'

class AndroidApp

  def initialize driver
    @driver = driver
    @appCategoryXpath = "//android.widget.TextView[@content-desc='App']"
    @activityCategoryXpath = "//android.widget.TextView[@content-desc='Activity']"
  end

  def findActivityCategory
    @driver.find_element(:xpath, @appCategoryXpath).click
    return @driver.find_element(:xpath, @activityCategoryXpath).text
  end

end
